Add the MeshGenerator system

leorachapuis opened the pull request:
This is the new PR for the MeshGenerators (the old one was number #11640 ).
This should be nicer because we fixed some issues with old commits.

friedmud committed:
Add MeshGenerators System refs #11640

friedmud committed:
Make sure that MeshGenerator chaining works refs #11640

friedmud committed:
Add FileMeshGenerator refs #11640

leorachapuis committed:
Move MeshModifers and Meshes into MeshGenerators refs #11640

Refs #11640

Create AnnularMeshGenerator.

Add in a dynamic_pointer_cast for unique_ptrs to make things a bit simpler for #11640

Update AnnularMeshGenerator. Refs #11640.

Generalize ElementGenerator.

Also create test for this class.
Refs #11640.

[WIP] Create MeshExtruderGenerator. Refs #11640.

Create RinglebMeshGenerator.
Also create a test for this class.
Refs #11640.

[WIP] Create SmoothMeshGenerator. Refs #11640.

Create SpiralAnnularMeshGenerator.
Also create a test for this class.
Refs #11640.

[WIP] Create StitchMeshGenerator. Refs #11640.

[WIP]Fix the cast problem in StitchedMeshGenerator.
std::vector<MeshGeneratorName> still not supported. Refs #11640.

Fix CastUniquePointer. Refs #11640.

[WIP] Fix SmoothMeshGenerator. Refs #11640.

Add ConcentricCircleMeshGenerator.
Refs #11640.

Add GeneratedMeshGenerator.
Also create a basic mesh test for this class.
Refs #11640.

Add DistributedGeneratedMeshGenerator.
Also create a basic mesh test for this class.
Refs #11640.

[WIP] create PatternedMeshGenerator.

[WIP] Create TiledMeshGenerator.
There's still a problem with the boundaries.
Also create a basic mesh test.
Refs #11640.

Move the function changeID from MooseMesh.

 Create ImageMeshGenerator.
Also create a basic mesh test.
Refs #11640.

Starting to add documentation for MeshGenerators. Refs #11640.

Fix MeshExtruderGenerator.
Now it works.Also create a basic mesh test for this class
Refs #11640.

Move the method getBoundaryIDs from MooseMesh.
It's now usable with MeshBase meshes.
Also add the possibility to have std::vector<MeshGeneratorName> in an input file.
Refs #11640.

Fix StitchedMeshGenerator. Refs #11640.

Clean up the code. Refs #11640.

Fix a mistake : template types were in the wrong order.
Refs #11640.

Fix PatternedMeshGenerator. Refs #11640.

Fix some mistakes linkes to CastUniquePointer.
Refs #11640.

Fix some comments and errors.

Using the old gold and input files to prove that the new capability works correctly.

Create GenerateSideSetsBase (intermediate base class).
Refs #11640.

Create GenerateAllSideSetsByNormals.
Refs #11640.

Create GenerateExtraNodeset.
Refs #11640.

Add basic mesh tests for GenerateAllSdeSetsByNormals and GenerateExtraNodeset. Refs #11640.

Create GenerateSideSetsFromBoundingBox.
Refs #11640.

Create GenerateElementSubdomainID.
Refs #11640.

Add quick tests for GenerateElementSubdomainID and GenerateSideSetsFromBoundingBox. Refs #11640.

Create GenerateSubdomainID.
Also create a basic test for this class.
Refs #11640.

Create the intermediate base class ElementDeletionBaseGenerator. Refs #11640.

Create BlockDeletionGenerator.
Also add a basic test for this class.
Refs #11640.

Create GenerateBoundingBoxNodeSet.
Also create a basic test.
Refs #11640.

Create the intermediate base class BreakMeshByBlockBaseGenerator. Refs #11640.

Create BreakMeshByBlockGenerator.
Also create a mesh test for this class.
Refs #11640.

Create LowerDBlockFromSidesetGenerator.
Also create a mesh test.
Refs #11640.

Create SubdomainBoundingBoxGenerator.
Also create a mesh test for this class.
Refs #11640.

Create OrientedSubdomainBoundingBoxGenerator.
Also add a quick mesh test.
Refs #11640.

Create ParsedSubdomainMeshGenerator.
Also add a mesh test for this class.
Refs #11640.

Create ParsedGenerateSideset.
Also create a basic test for this class.
Refs #11640.

Renaming the base classes properly. Refs #11640.

Create RenameBlockGenerator.
Also add a mesh test.
Refs #11640.

Update a test file

Create SideSetsAroundSubdomainGenerator.
Also add 2 mesh tests for this class.
Refs #11640.

Add a function based on MooseMesh::getSubdomainIDs.

Create SideSetsBetweenSubdomainsGeerator.
Also add 2 mesh tests.
Refs #11640.

Create BreakBoundaryOnSubdomainGenerator.
Also add a mesh test for this class.
Refs #11640.

Create MeshSideSetGenerator.
Also add a basic mesh test.
Refs #11640.

Create SideSetsFromNormalsGenerator.
Add a mesh test for this class.
Refs #11640.

Create SideSetsFromPointsGenerator.
Also create a mesh test.
Refs #11640.

Create TransformGenerator.
Also create a mesh test.
Refs #11640.

Create MeshBaseImageSampler and ImageSubdomainGenerator.
Also add 2 mesh tests for ImageSubdomainGenerator.
Refs #11640.

Add an include in ImageMeshGenerator. Refs #11640.

Add basic documentation files for some MeshGenerators.
Refs #11640.

Add basic documentation for the rest of the MeshGenerators. Refs #11640

Add a comment in MeshBaseImageSampler.C

Complete the test files and update some input files.

Fix some errors. Refs #11640.

Fix errors. Refs #11640

Apply clang-format. Refs #11640.

Include libmesh/mesh_base.h in MooseApp.h

Add some missing .e files in the test directories. Refs #11640.

Include MooseEnum.h in GeneratedMeshGenerator.h

Include MoodApp.h in MeshGeneratorMesh.h. Refs #11640.

Include replicated_mesh.h in PatternedMeshGenerator.h and StittchedMeshGenerator.h. Refs #11640.

Add includes in StitchedMeshGenerator.C and DistributedGeneratedMesh.h. Refs #11640.

Add the requirement block in 2 test files.

Fix some errors in the test files.. Refs #11640.

Fix some mistakes in AnnularMeshGenerator.C. Refs #11640.

Change the return statement in all the files. Refs #11640.

Fix a spelling mistake. Refs #11640.

Add some missing includes. Refs #11640.

Add some missing includes. Refs #11640.

Add a missing include in PatternedMeshGenerator.C

Update an exodus file. Refs #11640.

Enable vtk in the image_subdomain_generator test file. Refs #11640.

Fix a memory error in MeshSideSetGenerator.C.
Refs #11640.

Update the test files to fix the Distributed Mode fails. Refs #11640.

Apply clang-format. Refs #11640.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/251216/) on 0940402 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/12409/site)

This comment will be updated on new commits.

leorachapuis committed:
Trying to fix the failing test about DistributedGeneratedMeshGenerator. Refs #12409.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/251420/) on 6ceed37 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/12409/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/12409/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 094bddda303168aee41e90defc7dddde5b66783d`

leorachapuis committed:
Make DistributedGeneratedMeshGenerator not registered for the moment.
Refs #12409.

leorachapuis committed:
Applay clang-format. Refs #12409.

leorachapuis committed:
Comment some lines in DistributedGeneratedMeshGenerator.md. Refs #12409.

friedmud committed:
Turn off recover for mesh-only tests refs #11640

friedmud commented on framework/doc/content/source/mesh_generators/AnnularMeshGenerator.md:
```diff
@@ -0,0 +1,34 @@
+
+The number of elements in the radial direction and the angular direction may be specified. In addition, a growth factor on the element size in the radial direction may be chosen. The element-size (in the radial direction) is multiplied by this factor for each concentric ring of elements, moving from the inner to the outer radius.
+
+Sidesets are also created:
```
```suggestion
Sidesets created:
```

aeslaughter replied:
```suggestion
Sidesets are created:
```

friedmud commented on framework/doc/content/source/actions/AddMeshGeneratorAction.md:
```diff
@@ -0,0 +1,9 @@
+!alert construction title=Undocumented Class
+The AddMeshGeneratorAction has not been documented, if you would like to contribute to MOOSE by writing
+documentation, please see [/generate.md]. The content contained on this page explains the typical
+documentation associated with an action; however, what is contained is ultimately determined by what
```
```suggestion
stuff however, what is contained is ultimately determined by what
```

friedmud committed:
Turn off debug testing of json output refs #11640

friedmud committed:
Update AddMeshGeneratorAction.md

friedmud committed:
Update FileMeshGenerator.md

friedmud committed:
Rename mesh_generators->meshgenerators refs #11640

Rename many MeshGenerators refs #11640

Rename test directories #11640

Rename documentation refs #11640

Fixup doco refs #11t640

More renaming refs #11640

More renaming in tests refs #11640

Turn off test in debug mode refs #11640

moosebuild commented:
Job [Application Documentation](https://civet.inl.gov/job/252123/) on 76c2001 : invalidated by @friedmud

friedmud commented:
Started up many more of the optional tests - this adds so much code that I want to make sure it's tested well

moosebuild commented:
Job [Test timings](https://civet.inl.gov/job/252611/) on bdb9907 wanted to post the following:

View timings [here](http://mooseframework.inl.gov/docs/PRs/12409/timing)

This comment will be updated on new commits.

moosebuild commented:
Job [Test Parallel debug](https://civet.inl.gov/job/252602/) on bdb9907 : invalidated by @friedmud

friedmud commented:
@permcody @andrsd @cpgr Whatever is wrong with RELAP-7's documentation has nothing to do with this PR.  I think I accidentally added that set of tests to this PR (I've now turned it off).

friedmud reviewed (dismissed):
This is good - @leorachapuis has more work on this coming soon and we can change things over time too.

moosebuild commented:
Job [Application Documentation](https://civet.inl.gov/job/254035/) on 24cd49f : invalidated by @friedmud

moosebuild commented:
Job [Minimum clang version](https://civet.inl.gov/job/253745/) on 24cd49f : invalidated by @friedmud

moosebuild commented:
Job [Minimum clang version (debug)](https://civet.inl.gov/job/253749/) on 24cd49f : invalidated by @friedmud

moosebuild commented:
Job [Minimum gcc version](https://civet.inl.gov/job/253748/) on 24cd49f : invalidated by @friedmud

moosebuild commented:
Job [Minimum gcc version (debug)](https://civet.inl.gov/job/253750/) on 24cd49f : invalidated by @friedmud
